x-pack/platform/plugins/private/reporting/server/services/scheduled_reports/scheduled_query.ts

```diff
@@ -306,5 +319,135 @@ export function scheduledQueryFactory(reportingCore: ReportingCore): ScheduledQu
         });
       }
     },
+
+    async bulkDelete(logger, req, res, ids, user) {
```

Comment:
I think the function is too large, doing so many things. Could you please refactor a bit to make it easier to read? We can move a lot of the functionality to functions (inside the same folder) and reduce the code to perform a couple of steps. Wdyt?